lib/default.nix

```diff
@@ -9,85 +16,118 @@ let
+        inherit internalCrateNameForCleanSource;
+      };
+      cleanCargoToml = callPackage ./cleanCargoToml.nix { };
+      configureCargoCommonVarsHook = callPackage ./setupHooks/configureCargoCommonVars.nix { };
+      configureCargoVendoredDepsHook = callPackage ./setupHooks/configureCargoVendoredDeps.nix { };
+      devShell = callPackage ./devShell.nix { };
 
-  crateRegistries = self.registryFromDownloadUrl {
-    dl = "https://static.crates.io/crates";
-    indexUrl = "https://github.com/rust-lang/crates.io-index";
-  };
+      crateNameFromCargoToml = callPackage ./crateNameFromCargoToml.nix {
+        inherit internalCrateNameFromCargoToml;
+      };
 
-  downloadCargoPackage = callPackage ./downloadCargoPackage.nix { };
-  downloadCargoPackageFromGit = callPackage ./downloadCargoPackageFromGit.nix { };
-  filterCargoSources = callPackage ./filterCargoSources.nix { };
-  findCargoFiles = callPackage ./findCargoFiles.nix { };
-  inheritCargoArtifactsHook = callPackage ./setupHooks/inheritCargoArtifacts.nix { };
-  installCargoArtifactsHook = callPackage ./setupHooks/installCargoArtifacts.nix { };
-  installFromCargoBuildLogHook = callPackage ./setupHooks/installFromCargoBuildLog.nix { };
-  mkCargoDerivation = callPackage ./mkCargoDerivation.nix { };
-  mkDummySrc = callPackage ./mkDummySrc.nix { };
+      crateRegistries = self.registryFromDownloadUrl {
+        dl = "https://static.crates.io/crates";
+        indexUrl = "https://github.com/rust-lang/crates.io-index";
+      };
 
-  overrideToolchain = toolchain: self.overrideScope (_final: _prev: {
-    cargo = toolchain;
-    clippy = toolchain;
-    rustc = toolchain;
-    rustfmt = toolchain;
-  });
+      downloadCargoPackage = callPackage ./downloadCargoPackage.nix { };
+      downloadCargoPackageFromGit = callPackage ./downloadCargoPackageFromGit.nix { };
+      filterCargoSources = callPackage ./filterCargoSources.nix { };
+      findCargoFiles = callPackage ./findCargoFiles.nix { };
+      inheritCargoArtifactsHook = callPackage ./setupHooks/inheritCargoArtifacts.nix { };
+      installCargoArtifactsHook = callPackage ./setupHooks/installCargoArtifacts.nix { };
+      installFromCargoBuildLogHook = callPackage ./setupHooks/installFromCargoBuildLog.nix { };
+      mkCargoDerivation = callPackage ./mkCargoDerivation.nix { };
+      mkDummySrc = callPackage ./mkDummySrc.nix { };
 
-  path = callPackage ./path.nix {
-    inherit internalCrateNameForCleanSource;
-  };
+      overrideToolchain = toolchainArg: self.overrideScope (_final: _prev:
+        let
+          toolchain =
+            if lib.isFunction toolchainArg
+            then spliceToolchain toolchainArg
+            else toolchainArg;
+          needsSplicing = stdenv.buildPlatform != stdenv.hostPlatform && toolchain?__spliced == false;
+        in
+        lib.warnIf needsSplicing "crane overrideToolchain requires a spliced toolchain when cross-compiling" {
```

Comment:
I think the warning is good to keep! There can be a lot of subtle errors when setting up cross compilation, if we can catch a possible bad situation and at least warn on it it would be helpful to users (plus its costs us very little to have a warning here so I'm all for it)

Reply:
Though we may want to tweak the warning to hint towards the solution (namely, use a function instead of passing in a value directly), esp since this used to require an explicit value and anyone used to the older format might not understand what changed

Reply:
Reworded this comment!